<launch>
	<param name="/use_sim_time" value="true"/>

	<node name="listener" pkg="tf_listener" type="tf_listener.py" args="$(arg output)" />
	
	<node name="hector_mapping" pkg="hector_mapping" type="hector_mapping" output="screen" >
		<!-- Frame names -->
		<param name="map_frame" value="map" />
		<param name="base_frame" value="base_link" />
		<param name="odom_frame" value="odom" />

		<!-- Topic name -->
		<param name="scan_topic" value="base_scan" />

		<!-- Tf use -->
	    <param name="use_tf_scan_transformation" value="true"/>
	    <param name="use_tf_pose_start_estimate" value="false"/>
	    <param name="pub_map_odom_transform" value="true"/>
	    <param name="advertise_map_service" value="true"/>
	    <param name="map_with_known_poses" value="false"/>

	    <!-- Map size / start point -->
	    <param name="map_resolution" value="0.01"/>
	    <param name="map_size" value="4096"/>
	    <param name="map_start_x" value="0.5"/>
	    <param name="map_start_y" value="0.5" />
	    <param name="map_multi_res_levels" value="3" />

	    <!-- Map update parameters -->
	    <param name="update_factor_free" value="0.4"/>
	    <param name="update_factor_occupied" value="0.9" />    
	    <param name="map_update_distance_thresh" value="0.1"/>
	    <param name="map_update_angle_thresh" value="0.06" />
	</node>

	<node name="player" pkg="rosbag" type="play" args="--clock $(arg file)"/>
	
</launch>